Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h

```diff
@@ -634,7 +636,7 @@ namespace internal {
       BOOST_FOREACH(edge_descriptor e, edges(mesh_))
       {
         double sqlen = sqlength(e);
-        if( (sqlen < sq_low) && is_collapse_allowed(e) )
+        if( (sqlen < sq_low) && is_collapse_allowed(e) ) //SL I wonder if we shouldn't delay the is_collapse_allowed test
```

Comment:
it is re-tested in the `while (!short_edges.empty())` loop, so it is an option, indeed